server/Calculator.cpp

```diff
@@ -4,19 +4,29 @@
 #include <stdexcept>
 #include <regex>
 
-
 using namespace std;
 
 string Calculator::calculateExpression(string inExpr)
 {
-   //Parse input expression into tokens using spaces and operators as delimiters
-   vector<string> expTokens = expTokenize(inExpr);
+   //Validate input expression (i.e. containing valid characters)
+   set<char> validChars = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
```

Comment:
Should we also include A, B, C, D, E, F for the hex expression for the validChars as well?

Reply:
@troy8chen, this is the set for calculate expression function only. For your purposes, you will create a similar set with the chars that are valid for your specific function and pass it in as a parameter to the validate function